lib/tom_queue/deferred_work_manager.rb

```diff
@@ -113,6 +121,24 @@ def stop
       deferred_set && deferred_set.interrupt
     end
 
-  end
+    # Register handlers to shut down the manager gracefully
+    def register_signal_handlers
+      Thread.main[:signals] = []
+
+      [:QUIT, :INT, :TERM].each do |sig|
+        trap(sig) do
+          Thread.main[:signals] << sig
```

Comment:
This won't cycle the loop - so you'll presumably end up waiting for up to 2 seconds for the loop to spin. Rather than faffing with Thread variables, you should just call `#stop` directly - this will mark `@shutdown` and interrupt the blocking `pop` call, so it will shutdown immediately.